pkg/controller/kieapp/deploy_ui_test.go

```diff
@@ -0,0 +1,110 @@
+	assert.Nil(t, err, "Error setting operator owner as CSV")
+
+	err = service.Create(context.TODO(), operator)
+	assert.Nil(t, err, "Error creating the Operator")
+
+	var url string
+	service.CreateFunc = func(ctx context.Context, obj runtime.Object) error {
+		if route, matched := obj.(*v1.Route); matched {
+			url = fmt.Sprintf("%s.apps.example.com", route.Name)
+			route.Spec.Host = url
+		}
+		return service.Client.Create(ctx, obj)
+	}
+	deployConsole(&Reconciler{Service: service}, operator)
+
+	updatedCSV := &operators.ClusterServiceVersion{}
+	err = service.Get(context.TODO(), types.NamespacedName{Name: csv.Name, Namespace: csv.Namespace}, updatedCSV)
+	assert.Nil(t, err, "Error fetching CSV from client")
+
+	link := getConsoleLink(updatedCSV)
+	assert.NotNil(t, link, "Found no console link in CSV")
+	assert.Equal(t, fmt.Sprintf("https://%s", url), link.URL, "The console link did not have the expected value")
+}
+
+func TestUpdateExistingLink(t *testing.T) {
+	service := test.MockServiceWithExtraScheme(&operators.ClusterServiceVersion{}, &appsv1.Deployment{}, &corev1.Pod{})
+
+	box := packr.New("CSV", "../../../deploy/catalog_resources/redhat/1.1.0")
+	bytes, err := box.Find("businessautomation-operator.1.1.0.clusterserviceversion.yaml")
+	assert.Nil(t, err, "Error reading CSV file")
+	csv := &operators.ClusterServiceVersion{}
+	err = yaml.Unmarshal(bytes, csv)
+	assert.Nil(t, err, "Error parsing CSV file")
+	csv.Spec.Links = append([]operators.AppLink{{Name: constants.ConsoleLinkName, URL: "some-bad-link"}}, csv.Spec.Links...)
+
+	err = service.Create(context.TODO(), csv)
+	assert.Nil(t, err, "Error creating the CSV")
+
+	box = packr.New("Operator", "../../../deploy")
+	bytes, err = box.Find("operator.yaml")
+	assert.Nil(t, err, "Error reading Operator file")
+	operator := &appsv1.Deployment{}
+	err = yaml.Unmarshal(bytes, operator)
+	assert.Nil(t, err, "Error parsing Operator file")
+
+	err = controllerutil.SetControllerReference(csv, operator, service.GetScheme())
+	assert.Nil(t, err, "Error setting operator owner as CSV")
+
+	err = service.Create(context.TODO(), operator)
+	assert.Nil(t, err, "Error creating the Operator")
+
+	var url string
+	service.CreateFunc = func(ctx context.Context, obj runtime.Object) error {
+		if route, matched := obj.(*v1.Route); matched {
+			url = fmt.Sprintf("%s.apps.example.com", route.Name)
+			route.Spec.Host = url
+		}
+		return service.Client.Create(ctx, obj)
+	}
+	deployConsole(&Reconciler{Service: service}, operator)
+
+	updatedCSV := &operators.ClusterServiceVersion{}
+	err = service.Get(context.TODO(), types.NamespacedName{Name: csv.Name, Namespace: csv.Namespace}, updatedCSV)
+	assert.Nil(t, err, "Error fetching CSV from client")
+
+	link := getConsoleLink(updatedCSV)
+	assert.NotNil(t, link, "Found no console link in CSV")
+	assert.Equal(t, fmt.Sprintf("https://%s", url), link.URL, "The console link did not have the expected value")
```

Comment:
instead of verifying against `link.URL`, maybe check the `updatedCSV.Spec.Links` map for the proper link instead? that's the object that will be will ultimately get updated in openshift

Reply:
That's what I'm doing. I'm just reusing the `getConsoleLink` function to look in `updatedCSV.Spec.Links` for the one link that I care about.